ICSharpCode.ILSpyX/MermaidDiagrammer/AssemblyInfo.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Diagnostics;
+using System.Reflection;
+
+namespace ICSharpCode.ILSpyX.MermaidDiagrammer
+{
+	internal static class AssemblyInfo
```

Comment:
Can we rename to something like MermaidDiagrammerVersionInfo? AssemblyInfo.cs (the name) tripped me up in thinking it is such a standard file.

Reply:
Can do - I remember stumbling over it as well coming back to it :)

This version info is actually used to **display the builder version** and **link to the associated readme** (requires github repo tags with the same version format!) marked in yellow here:
![image](https://github.com/user-attachments/assets/2bf2e74e-0f27-4ad1-908f-13a22d9e0388)

We should update that anyway to display info of either _ILSpyCmd_ or _ILSpyX_ (considering that's where the builder code lives and a UI adaptation is planned). Which builder info should I put there? That of the _Executing_ assembly or that of _ILSpyX_? And what (maybe versioned?) documentation links should the build info include?

Reply:
ILSpyX - because if somebody else embeds it, obviously the important information is what is in the library. Take a look at the ILSpy About page for how we obtain the underlying version information.

Reply:
> Take a look at the ILSpy About page for how we obtain the underlying version information.

https://github.com/icsharpcode/ILSpy/blob/3746337d376584ac6dec7d8672dbdd034046d8c9/ILSpy/AboutPage.cs#L70

Do you mean `DecompilerVersionInfo.FullVersionWithCommitHash` ? That creates a really long and detailed string for a for a user-facing label. Compare:
v9.0.0 (old)
v9.0.0.7848-mermaid-diagrammer-preview3+ceca662c8bef1db6b018d00b52a83d11c8a47921 (new)

Also, that won't work to link to versioned docs. But looking at the tags in this repo, I'm not sure whether automatically versioned doco links are possible - since the tag names don't follow a pattern predictable at compile time.

Should the link then always go to the master branch version of doco? And are you sure you want the commit hash in the builder version info of the generated diagrammer?

Reply:
That string is shorter on main builds, eg the branch info is not included. The reason for including all this: it allows for proper tracking of bug reports. The fourth part of the version does not allow to easily go back to the commit hash, and that is why we started including it.

Reply:
I see and I get that - tagging builds with the commit hash is a good practice I use myself. Never the full hash tho -  the first 6 or 8 chars should be enough, especially with the version prefix. And especially for a user-facing display, you want as little noise as possible. But I'll display `DecompilerVersionInfo.FullVersionWithCommitHash` for now to have at least a uniform display.